src/profile-logic/import/instruments/BinaryPlistParser.js

```diff
@@ -0,0 +1,221 @@
+
+    // Just use the last 32-bits of these 64-bit big-endian values
+    const objectCount = this.view.getUint32(trailer + 12, false);
+    const rootIndex = this.view.getUint32(trailer + 20, false);
+    let tableOffset = this.view.getUint32(trailer + 28, false);
+
+    // Parse all offsets before starting to parse objects
+    for (let i = 0; i < objectCount; i++) {
+      this.offsetTable.push(this.parseInteger(tableOffset, offsetSize));
+      tableOffset += offsetSize;
+    }
+
+    return this.parseObject(this.offsetTable[rootIndex]);
+  }
+
+  parseLengthAndOffset(offset: number, extra: number) {
+    if (extra !== 0x0f) return { length: extra, offset: 0 };
+    const marker = this.view.getUint8(offset++);
+    if ((marker & 0xf0) !== 0x10) {
+      throw new Error('Unexpected non-integer length at offset ' + offset);
+    }
+    const size = 1 << (marker & 0x0f);
+    return { length: this.parseInteger(offset, size), offset: size + 1 };
+  }
+
+  parseSingleton(offset: number, extra: number): any {
+    if (extra === 0) {
+      return null;
+    }
+    if (extra === 8) {
+      return false;
+    }
+    if (extra === 9) {
+      return true;
+    }
+    throw new Error('Unexpected extra value ' + extra + ' at offset ' + offset);
+  }
+
+  parseInteger(offset: number, size: number): number {
+    if (size === 1) {
+      return this.view.getUint8(offset);
+    }
+    if (size === 2) {
+      return this.view.getUint16(offset, false);
+    }
+    if (size === 4) {
+      return this.view.getUint32(offset, false);
+    }
+
+    if (size === 8) {
+      return (
+        Math.pow(2, 32 * 1) * this.view.getUint32(offset + 0, false) +
+        Math.pow(2, 32 * 0) * this.view.getUint32(offset + 4, false)
+      );
+    }
+
+    if (size === 16) {
+      return (
+        Math.pow(2, 32 * 3) * this.view.getUint32(offset + 0, false) +
+        Math.pow(2, 32 * 2) * this.view.getUint32(offset + 4, false) +
+        Math.pow(2, 32 * 1) * this.view.getUint32(offset + 8, false) +
+        Math.pow(2, 32 * 0) * this.view.getUint32(offset + 12, false)
+      );
+    }
+
+    throw new Error(
+      'Unexpected integer of size ' + size + ' at offset ' + offset
+    );
+  }
+
+  parseFloat(offset: number, size: number): number {
+    if (size === 4) {
+      return this.view.getFloat32(offset, false);
+    }
+    if (size === 8) {
+      return this.view.getFloat64(offset, false);
+    }
+    throw new Error(
+      'Unexpected float of size ' + size + ' at offset ' + offset
+    );
+  }
+
+  parseDate(offset: number, size: number): Date {
+    if (size !== 8) {
+      throw new Error(
+        'Unexpected date of size ' + size + ' at offset ' + offset
+      );
+    }
+    const seconds = this.view.getFloat64(offset, false);
+    return new Date(978307200000 + seconds * 1000); // Starts from January 1st, 2001
+  }
+
+  parseData(offset: number, extra: number): Uint8Array {
+    const both = this.parseLengthAndOffset(offset, extra);
```

Comment:
what does that mean, `both` ? This really isn't clear to me what this return value represents compared to the parameters. (same applies below)